### PR TITLE
fix: Better printing of XL config init error

### LINKED ISSUE #4831: minio cluster startup with a error message showing unformatted disk found
Hi!
I have a 4 nodes minio cluster, one node is 10 seconds apart from the other 3 nodes(I know it shouldn't be bigger than 3 seconds apart).When I start the minio server,the console print some error message: 'unformatted disk found'.When I use ntp to make the time right,the error message is gone and the cluster runs perferctly.So maybe the error message can change a little bit.

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
When there is a configuration error, exit immediately and show a simple error message.

time="2017-12-08T23:25:36+01:00" level=error msg="Initializing object layer failed" cause="http://192.168.100.143:9002/tmp/dist2: Authentication failed, check your access credentials" source="[server-main.go:211:serverMain()]"

## Motivation and Context
Fixes #4831 

## How Has This Been Tested?
go test + Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -132,10 +133,13 @@ func isMaxPartID(partID int) bool {
 	return partID > globalMaxPartID
 }
 
-func contains(stringList []string, element string) bool {
-	for _, e := range stringList {
-		if e == element {
-			return true
+func contains(slice interface{}, elem interface{}) bool {
```
Comment (krishnasrinivas) [inline cmd/utils.go]: is it being used for anything other than string? if not it is better to keep it as string.

**Review by krishnasrinivas — DISMISSED** [review]
lgtm, tested:
```
krishna@escape:~$ server4
ERRO[0000] Initializing object layer failed              cause=http://localhost:9001/export1: Authentication failed, check your access credentials source=[server-main.go:211:serverMain()]
krishna@escape:~$ 

```

**Comment by krishnasrinivas** [comment]
@vadmeste but the other servers start printing:
```
Initializing data volume for first time. Waiting for other servers to come online (elapsed 12m0s)
ERRO[0728] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
Initializing data volume for first time. Waiting for other servers to come online (elapsed 12m8s)
ERRO[0754] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
Initializing data volume for first time. Waiting for other servers to come online (elapsed 12m34s)
ERRO[0762] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
Initializing data volume for first time. Waiting for other servers to come online (elapsed 12m42s)
ERRO[0792] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
Initializing data volume for first time. Waiting for other servers to come online (elapsed 13m12s)
ERRO[0808] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
Initializing data volume for first time. Waiting for other servers to come online (elapsed 13m28s)
ERRO[0837] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
Initializing data volume for first time. Waiting for other servers to come online (elapsed 13m57s)
```
can the following error not be printed:
```
ERRO[0837] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]
```

**Comment by kannappanr** [comment]
@vadmeste Other than what @krishnasrinivas pointed out. I am fine with what is being printed

**Comment by vadmeste** [comment]
@krishnasrinivas I added a test for contains.



> @vadmeste but the other servers start printing:
> 
> Initializing data volume for first time. Waiting for other servers to come online (elapsed 12m0s)
> ERRO[0728] Disk http://localhost:9004/export4 is still unreachable  cause=disk not found source=[prepare-storage.go:222:printRetryMsg()]

This is not really related to this PR. It has been like that for a long time. This is shown only after some number of initialization  retries.. I added it to show the real reason why the cluster wasn't able to initialize itself. I can enhance it in a separate PR.

**Comment by vadmeste** [comment]
ping @krishnasrinivas @kannappanr

**Comment by kannappanr** [comment]
@vadmeste overall it looks ok. But, combineDiskErrs is being used in another place in prepare-storage-message.go, 

```// Generate a formatted message when cluster is misconfigured.
func getConfigErrMsg(storageDisks []StorageAPI, sErrs []error) string {
	msg := colorBlue("\nDetected configuration inconsistencies in the cluster. Please fix following servers.")
	return msg + combineDiskErrs(storageDisks, sErrs)
}```

Should this also be modified to use `reduceInitXLErrs` instead?

**Comment by kannappanr** [comment]
@vadmeste can you add unit test for `reduceInitXLErrs`

**Comment by vadmeste** [comment]
done @kannappanr

**Review by krishnasrinivas — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]

**Pull request merged**